src/Behat/Context/EntityContext.php

```diff
@@ -355,11 +355,27 @@ public function entityCreate($entity_type, $entity) {
   /**
    * @AfterScenario
    */
-  public function deleteGeneratedEntities() {
-    foreach ($this->entities as $entity_type => $ids) {
-      $this->getCore()->entityDeleteMultiple($entity_type, $ids);
+  public function cleanEntities() {
+
+    $bypass_entities = isset($this->customParameters['entities_clean_bypass']) ? $this->customParameters['entities_clean_bypass'] : FALSE;
```

Comment:
This should not be neccesary. The entities generated in each scenario should be deleted. Why is this neccesary?

Reply:
@rsanzante  if the order matters, why don't sort the entities before deleting? We could create a static array with the mappins like this:

[
  group: 1
  default: -1,
]

With the usort function we may order the entities to delete first group and then the rest of them.

Reply:
Plus, this configuration would be overrwritten in behat.yml.

Reply:
Just some background info about the nature of the problem, which has two sides.

1. The sorting won't work in this case because groups and subgroups aren't separate entities, they are both groups "tied" by an intermediate entity (group_content) that acts much like an intermediate table does in sql. 

In Group entities, therefore, the deletion order matters a lot (i.e. a subgroup must always be **explicitly** deleted before a group). The intent to extend it to the whole library comes from the idea that "last in, first out" is a better default deletion order than "first in, first out" as we expect dependent entities to be always created before their parents. 

The sorting sceneario *could only work* if it accepted bundles, i.e:

```php
[
group:sel_process:1
group:company:2
default: -1,
]


2. There is a second problem. There is an "intermediate entity" (group_content) that, in Groups, is **almost never** deleted manually but as a result of group deletion. So, this is the opposite scenario: there is a dependency but in this case **we don't want group content to be deleted before the parent** but let the parent automatically remove the entities.

This is something that a sort could solve, because you would need only to make sure that groups are deleted before and, when group_content comes, they will be already wiped.